.github/workflows/check_docker.yml

```diff
@@ -26,15 +34,42 @@ jobs:
           grep -qE "writing image.* done" /tmp/build.log || { echo "Build failed - no success pattern"; cat /tmp/build.log; exit 1; }
 
           if ! command -v nvidia-ctk >/dev/null 2>&1; then
-            sed -i 's/check_nvidia_ctk()/pass  # nvidia-ctk not available/g' utilities/cli/container.py
-            sed -i 's/"--runtime",/# "--runtime",/g' utilities/cli/container.py
-            sed -i 's/"nvidia",/# "nvidia",/g' utilities/cli/container.py
-            sed -i 's/"--gpus",/# "--gpus",/g' utilities/cli/container.py
-            sed -i 's/"all",/# "all",/g' utilities/cli/container.py
+            eval "$CPU_CI_COMMANDS"
           fi
 
           ./holohub run-container --base-img ubuntu:${ubuntu_version} --no-docker-build | grep -q "docker run" || { echo "run-container failed"; exit 1; }
           ./holohub run-container --docker-opts "--memory 4g" --no-docker-build | grep -q "memory 4g" || { echo "docker-opts test failed"; exit 1; }
           ./holohub run-container --no-docker-build --add-volume "/tmp" | grep -q "/tmp" || { echo "add-volume test failed"; exit 1; }
           ./holohub run-container --no-docker-build -- echo hello > /tmp/trailing-args.log 2>&1
           grep -q "hello" /tmp/trailing-args.log || { echo "trailing args test failed"; cat /tmp/trailing-args.log; exit 1; }
+
+  test-entrypoint-images:
+    runs-on: ubuntu-22.04
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - image: "alpine:latest"  # CMD ["/bin/sh"]
+          - image: "busybox:latest" # CMD ["sh"]
+          - image: "jenkins/jenkins:latest"    # ENTRYPOINT ["/usr/local/bin/jenkins.sh"]
+          - image: "nginx:latest"   # ENTRYPOINT ["/docker-entrypoint.sh"] CMD ["nginx", "-g", "daemon off;"]
```

Comment:
Slight concern about relying on these for test coverage, we don't typically use these images and maintenance effort could be disproportionate if one of them decides to shift their entrypoint strategy upstream, even if unlikely.

Would it be difficult to write our own minimal Dockerfiles to cover these cases? Could either define in shell steps here or check in as test files.

Reply:
updated to use basic dockerfile to simulate the use cases